rules/jsonapi-document-structure-resource-fields.js

```diff
@@ -0,0 +1,85 @@
+
+// Errors - Processing Error  - https://jsonapi.org/format/#document-resource-object-fields
+
+// All rules in the file MUST have corresponding tests
+
+import { falsy } from '@stoplight/spectral-functions';
+
+export default {
+  documentationUrl: 'https://jsonapi.org/format/#document-resource-object-fields',
+  rules: {
```

Comment:
The rules you have created are individually targeting each area where we want to enforce the specifications for JSON:API v1.0 DocumentStructure.ResourceObjects.Fields.

However, you are currently only targeting the specifications where a `Resource Object` is an array. Per JSON:API v1.0 DocumentStructure.TopLevel Specifications for `Resource Objects`, the primary data **MUST** be either:

- a single `resource object`, a single `resource identified object`, or null, for a requests that target single resources
- An array of `resource objects`, an array of `resource identifier objects`, or an empty array, for requests that target resource collections. 

Your current ruleset does not target all of these possibilities where we must validate against. You must either adjust your current `JSONPath Expressions` to consider these possibilities or create additional rules to target these specifications per JSON:API v1.0.